libvast/src/system/explorer.cpp

```diff
@@ -134,23 +135,23 @@ explorer(caf::stateful_actor<explorer_state>* self, caf::actor node,
         VAST_DEBUG(self, "could not find timestamp field in", layout);
         return;
       }
-      std::optional<table_slice::column_view> by_column;
+      std::optional<table_slice_column> by_column;
       if (st.by) {
         // Need to pivot from caf::optional to std::optional here, as the
         // former doesnt support emplace or value assignment.
-        if (auto vopt = slice->column(*st.by))
-          by_column.emplace(*vopt);
+        if (auto col = table_slice_column{slice, *st.by}; col.slice())
+          by_column.emplace(std::move(col));
         if (!by_column) {
           VAST_TRACE("skipping slice with", layout, "because it has no column",
                      *st.by);
           return;
         }
       }
       VAST_DEBUG(self, "uses", it->name, "to construct timebox");
-      auto column = slice->column(it->name);
-      VAST_ASSERT(column);
-      for (size_t i = 0; i < column->rows(); ++i) {
-        auto data_view = (*column)[i];
+      auto column = table_slice_column{slice, it->name};
+      VAST_ASSERT(column.slice());
```

Comment:
This is a strange way to test for validity. Imho value types must be valid by construction.